harness/determined/exec/launch.py

```diff
@@ -2,13 +2,30 @@
 import json
 import logging
 import os
+import signal
 import subprocess
 import sys
+import types
 from typing import Dict
 
 import determined as det
 import determined.common
-from determined.common import constants, storage
+from determined.common import api, constants, storage
+from determined.exec import prep_container
+
+
+# Signal handler to intercept SIGTERM notification of pending preemption
+def trigger_preemption(signum: int, frame: types.FrameType) -> None:
```

Comment:
@rb-determined-ai FYI, mostly just that this exists for slurm (which preempts us with SIGTERM).